go-controller/pkg/cluster/master.go

```diff
@@ -0,0 +1,143 @@
+package cluster
+
+import (
+	"fmt"
+	"net"
+	"os/exec"
+
+	"github.com/golang/glog"
+
+	utilwait "k8s.io/apimachinery/pkg/util/wait"
+	kapi "k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/openshift/origin/pkg/util/netutils"
+)
+
+func (cluster *OvnClusterController) StartClusterMaster(masterNodeName string) error {
+	clusterNetwork := cluster.ClusterIPNet
+	hostSubnetLength := cluster.HostSubnetLength
+
+	subrange := make([]string, 0)
+	existingNodes, err := cluster.Kube.GetNodes()
+	if err != nil {
+		glog.Errorf("Error in initializing/fetching subnets: %v", err)
+		return err
+	}
+	for _, node := range existingNodes.Items {
+		hostsubnet, ok := node.Annotations[OVN_HOST_SUBNET]
+		if ok {
+			subrange = append(subrange, hostsubnet)
+		}
+	}
+	masterSwitchNetwork, err := calculateMasterSwitchNetwork(clusterNetwork.String(), hostSubnetLength)
+	if err != nil {
+		return err
+	}
+	subrange = append(subrange, masterSwitchNetwork)
+	cluster.masterSubnetAllocator, err = netutils.NewSubnetAllocator(clusterNetwork.String(), hostSubnetLength, subrange)
+	if err != nil {
+		return err
+	}
+
+	// now go over the 'existing' list again and create annotations for those who do not have it
+	for _, node := range existingNodes.Items {
+		_, ok := node.Annotations[OVN_HOST_SUBNET]
+		if !ok {
+			err := cluster.addNode(&node)
+			if err != nil {
+				glog.Errorf("error creating subnet for node %s: %v", node.Name, err)
+			}
+		}
+	}
+
+	cluster.SetupMaster(masterNodeName, masterSwitchNetwork)
+
+	go utilwait.Forever(cluster.watchNodes, 0)
+	return nil
+}
+
+func calculateMasterSwitchNetwork(clusterNetwork string, hostSubnetLength uint32) (string, error) {
+	subAllocator, err := netutils.NewSubnetAllocator(clusterNetwork, hostSubnetLength, make([]string, 0))
```

Comment:
Since the functions being called do not have comments on what they do, can we add a comment here about what they do and what is the return value for both the called functions here?